### ISSUE TITLE
2.2.2 breaks compatibility with older Rubies

### ISSUE THREAD

**Issue by dnrce** [socketry/multipart-post#94]
2.2.2 dropped support for Ruby <2.6.0. This is a backwards-incompatible change, so per semver it should be a major version bump. Older Rubies (with older rubygems) depending on `multipart-post ~> 2.x` will now fail to install:
```
$ docker run --rm -it ruby:2.5-alpine gem install multipart-post -v '~> 2.2'
Fetching multipart-post-2.2.2.gem
ERROR:  Error installing multipart-post:
	The last version of multipart-post (~> 2.2) to support your Ruby & RubyGems was 2.2.0. Try installing it with `gem install multipart-post -v 2.2.0`
	multipart-post requires Ruby version >= 2.6.0. The current ruby version is 2.5.9.229.
```
I'd suggest that the current gem should be rereleased as 3.0, and 2.2.2 be yanked.

I recognize that 2.5 (and 2.6) are old Rubies at this point, so I have no overall complaint about dropping support in a newer gem version here. It's just that the way this was implemented, those older Rubies can't even keep going with version constraints that were previously valid.

Thanks!

**Comment by deivid-rodriguez** [comment]
Hi!

If I understood what happened correctly, the problem is that 2.2.0 does not work at all on Ruby 2.5 (it probably uses newer syntax), so support was effectively dropped at 2.2.0. However, due to an oversight, the gemspec was not updated to reflect that and thus installing 2.2.0 the gem is possible on Ruby 2.5. 2.2.2 corrects that.

By the way, this case should be better handled by newer `gem` CLI versions, because it will automatically choose a version compatible with your Ruby. Sadly, in this case, I don't think it's really compatible but at least you won't get a `gem install` error which might be your actual concern.

```
$ docker run --rm -it ruby:2.5-alpine sh -c "gem update --system && gem install multipart-post -v '~> 2.2'"
(...)
(a bunch of RubyGems upgrade output)
(...)
RubyGems system software updated
Fetching multipart-post-2.2.0.gem
Successfully installed multipart-post-2.2.0
1 gem installed
```

or

```
$ docker run --rm -it ruby:2.5-alpine sh -c "gem update --system && gem install faraday -v '~> 1.0'"
(...)
(a bunch of RubyGems upgrade output)
(...)
RubyGems system software updated
Fetching ruby2_keywords-0.0.5.gem
Fetching faraday-retry-1.0.3.gem
Fetching faraday-rack-1.0.0.gem
Fetching faraday-patron-1.0.0.gem
Fetching faraday-net_http_persistent-1.2.0.gem
Fetching faraday-1.10.0.gem
Fetching multipart-post-2.2.0.gem
Fetching faraday-multipart-1.0.4.gem
Fetching faraday-net_http-1.0.1.gem
Fetching faraday-httpclient-1.0.1.gem
Fetching faraday-excon-1.1.0.gem
Fetching faraday-em_synchrony-1.0.0.gem
Fetching faraday-em_http-1.0.0.gem
Successfully installed ruby2_keywords-0.0.5
Successfully installed faraday-retry-1.0.3
Successfully installed faraday-rack-1.0.0
Successfully installed faraday-patron-1.0.0
Successfully installed faraday-net_http_persistent-1.2.0
Successfully installed faraday-net_http-1.0.1
Successfully installed multipart-post-2.2.0
Successfully installed faraday-multipart-1.0.4
Successfully installed faraday-httpclient-1.0.1
Successfully installed faraday-excon-1.1.0
Successfully installed faraday-em_synchrony-1.0.0
Successfully installed faraday-em_http-1.0.0
Successfully installed faraday-1.10.0
13 gems installed
```

**Comment by ioquatix** [comment]
I agree this is a pain point and I'm open to ideas about how we can make this work.

The best advice I can give you if you are using EOL Ruby is to pin on `multipart-mime` "~> 2.1.0".

I would probably have to defer to @deivid-rodriguez 's advice because it makes sense that older Rubies should only install the most recent compatible version. Unfortunately v2.2.0 won't be compatible with older Rubies either (to the extent that it might be an issue) and therefore while it might install, there is no guarantee it would work.

**Comment by richardstrnad** [comment]
This also broke the `github pages` provider from travis-ci. Worked before the 2.2.2 release, but now no longer.

```
Installing deploy dependencies
ERROR:  Error installing dpl-pages:
	The last version of multipart-post (>= 1.2, < 3) to support your Ruby & RubyGems was 2.2.0. Try installing it with `gem install multipart-post -v 2.2.0` and then running the current command again
	multipart-post requires Ruby version >= 2.6.0. The current ruby version is 2.4.5.335.
```

**Comment by ioquatix** [comment]
We can try to retroactively support EOL Ruby versions, say back to 2.3, if the compatibility burden is small.

**Comment by ioquatix** [comment]
Okay, I'm comfortable supporting back to at least 2.3 as the changes are relatively minor:

https://github.com/socketry/multipart-post/pull/95

**Comment by ioquatix** [comment]
Okay this is released in v2.2.3 - hopefully we didn't break anything else.

**Comment by deivid-rodriguez** [comment]
Nice! 💯
